tools/diagnostics/create_service_diagnostics_bundle.py

```diff
@@ -0,0 +1,224 @@
+
+def preflight_check() -> (int, bool, dict):
+    args = parse_args()
+    package_name_given = args.package_name
+    service_name = args.service_name
+    bundles_directory = args.bundles_directory
+    should_prompt_user = not args.yes
+
+    (is_authenticated, message) = is_authenticated_to_dcos_cluster()
+    if not is_authenticated:
+        log.error(
+            "We were unable to verify that you're authenticated to a DC/OS cluster.\nError: %s",
+            message,
+        )
+        return (1, False, {})
+
+    (rc, cluster_or_error) = attached_dcos_cluster()
+    if rc != 0:
+        log.error(
+            "We were unable to verify the cluster you're attached to.\nError: %s",
+            cluster_or_error,
+        )
+        return (rc, False, {})
+
+    (cluster_name, dcos_version, cluster_url) = cluster_or_error
+
+    (rc, marathon_app_or_error) = get_marathon_app(service_name)
+    if rc == 0:
+        package_name = marathon_app_or_error.get("labels", {}).get("DCOS_PACKAGE_NAME")
+        package_version = marathon_app_or_error.get("labels", {}).get("DCOS_PACKAGE_VERSION")
+    else:
+        log.info(
+            "We were unable to get details about '%s'.\nIssue: %s",
+            service_name,
+            marathon_app_or_error,
+        )
+        log.info(
+            "Maybe the '%s' scheduler is not running. That's ok, we can still try to fetch any "
+            + "artifacts related to it",
+            service_name,
+        )
+        package_name = package_name_given
+        package_version = "n/a"
+
+    if package_name_given != package_name:
+        log.error(
+            "Package name given '%s' is different than actual '%s' package name: '%s'",
+            package_name_given,
+            service_name,
+            package_name,
+        )
+        log.info("Try '--package-name=%s'", package_name)
+        return (1, False, {})
+
+    return (
+        0,
+        True,
+        {
+            "package_name": package_name,
+            "service_name": service_name,
+            "package_version": package_version,
+            "cluster_name": cluster_name,
+            "bundles_directory": bundles_directory,
+            "dcos_version": dcos_version,
+            "cluster_url": cluster_url,
+            "should_prompt_user": should_prompt_user,
+        },
+    )
+
+
+def main(argv):
+    rc, should_proceed, args = preflight_check()
+    if not should_proceed:
+        return rc
+
+    print("\nWill create bundle for:")
+    print("  Package:         {}".format(args.get("package_name")))
+    print("  Package version: {}".format(args.get("package_version")))
+    print("  Service name:    {}".format(args.get("service_name")))
+    print("  DC/OS version:   {}".format(args.get("dcos_version")))
+    print("  Cluster URL:     {}".format(args.get("cluster_url")))
+
+    if args.get("should_prompt_user"):
+        answer = input("\nProceed? [Y/n]: ")
+        if answer.strip().lower() in ["n", "no", "false"]:
```

Comment:
Why not `answer.strip().lower() not in ["yes", y"]:` instead?

Reply:
Because the default action is `Y`, indicated by its capitalization. Pressing enter with no actual input should proceed, for example. I guess this can be `if answer.strip().lower() not in ["yes", "y", ""]:`?

Reply:
The problem is that `"foo"`, `"nooooooo"` etc. will also be taken as `yes` --> I would rather check for the empty string explicitly then.

Reply:
f76163cf3d6f7c7e8000e5652520583cc5417aec